quantum_annealing.cpp

```diff
@@ -0,0 +1,474 @@
+//Implemented QuantumAnnealing algorithm has the following issue:
+//computeQuantumChangeScore has to be redesigned beacause
+//Total score !=  Sum of scores of individual pu change
+//due mutual effect via species cups, boundaries and clumping.
+
+
+#define DEBUGTRACEFILE
+#define PROB2D
+
+// Flags to change to "define" for debugging purposes
+#undef EXTRADEBUGTRACE
+#undef ANNEALING_TEST
+#undef DEBUGCHANGEPEN
+#undef DEBUGCALCPENALTIES
+#undef DEBUG_PRINTRESVALPROB
+#undef DEBUG_COUNTMISSING
+#undef DEBUG_HEURISTICS
+#undef DEBUG_IIMPHEUR
+#undef DEBUG_CLUSTERANALYSIS
+#undef DEBUG_CONNECTIONCOST
+#undef DEBUG_RESERVECOST
+#undef DEBUGCHECKCHANGE
+#undef DEBUG_CONNECTIONCOST2
+#undef CREATE_R_SCRIPTS
+#undef DEBUG_QA
+#undef DEBUG_PROB1D
+#undef DEBUG_PROB2D
+
+#include <algorithm>
+#include <chrono>
+#include <ctime>
+#include <cfloat>
+#include <iostream>
+#include <omp.h>
+#include <chrono> 
+
+// load the required function definition modules
+#include "utils.hpp"
+#include "algorithms.hpp"
+#include "computation.hpp"
+#include "clumping.hpp"
+#include "anneal.hpp"
+#include "heuristics.hpp"
+#include "probability.hpp"
+#include "input.hpp"
+#include "output.hpp"
+
+#include "score_change.hpp"
+
+namespace marxan {
+
+    using namespace algorithms;
+    using namespace utils;
+
+    double rQAPROP = 0.5, rQADECAY = 0.0001, rQADECAYB = 0, rQAACCPR = 0;
+    int iQADECAYTYPE = 0;
+
+    // determines if the change value for changing status for a set of planning units is good
+    // does it stochastically fall below the current acceptance probability?
+    int isGoodQuantumChange(struct scost change, double rProbAcceptance, uniform_real_distribution<double>& float_range, rng_engine& rngEngine)
+    {
+        if (change.total <= 0)
+            return 1;
+        else
+            return (rProbAcceptance > float_range(rngEngine)) ? 1 : 0;
+    }
+
+
+    // compute change in the objective function score for adding or removing a set of planning units
+    void computeQuantumChangeScore(int spno, int puno, const vector<spustuff>& pu, const vector<sconnections>& connections,
+        vector<sspecies>& spec, const vector<spu>& SM, vector<spu_out>& SM_out, const vector<int>& R, double cm,
+        scost& change, scost& reserve, double costthresh, double tpf1, double tpf2,
+        double timeprop, int clumptype, int iFluctuationCount, const vector<int>& PUChosen)
+        // imode = 1 add PU, imode = -1 remove PU
+    {
+
+        //computeQuantumChangeScore has to be redesigned beacause
+        //Total score !=  Sum of scores of individual pu change
+        //due mutual effect via species cups, boundaries and clumping.
+        throw runtime_error("computeQuantumChangeScore has to be redesigned beacause Total score !=  Sum of scores of individual pu change/n"); 
```

Comment:
I'm guessing we don't plan to fix this, so users will just get this error message?

Reply:
Could we let it run like before but just put a warning message?

Reply:
It is not hooked up right now. It is a break for a future developer who will not be able to go further before fixing code.